src/main/java/ch/njol/skript/lang/ExecutionIntent.java

```diff
@@ -0,0 +1,69 @@
+package ch.njol.skript.lang;
+
+import com.google.common.base.Preconditions;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Used to describe the intention of a {@link TriggerItem}.
+ * Currently only used to tell whether the item halts the execution or not and print the appropriate warnings.
+ * 
+ * @see TriggerItem#executionIntent() 
+ */
+public sealed interface ExecutionIntent extends Comparable<ExecutionIntent>
+	permits ExecutionIntent.StopTrigger, ExecutionIntent.StopSections {
+
+	static StopTrigger stopTrigger() {
+		return new StopTrigger();
+	}
+
+	static StopSections stopSections(int levels) {
+		Preconditions.checkArgument(levels > 0, "Depth must be at least 1");
+		return new StopSections(levels);
+	}
+
+	static StopSections stopSection() {
+		return new StopSections(1); 
+	}
+
+	@Nullable ExecutionIntent use();
+
+	final class StopTrigger implements ExecutionIntent {
```

Comment:
This was discussed in a vc, and we decided to keep them all as classes rather than records to force users to use the static methods rather than the constructors